src/services/infra/InfraService.ts

```diff
@@ -121,6 +125,15 @@ export default class InfraService {
     }
   }
 
+  isRootDomain = (primaryDomain: string) => {
+    // method to differentiate root domains with 4th level domains
+    if ((primaryDomain.match(/./g) || []).length < 3) {
+      // eg. blah.gov.sg
+      return true
+    }
+    return false
+  }
```

Comment:
This might not be right. Elaborating on my thought process:

We are checking if the given domain is a root domain because we want to make sure that we do not create a CNAME record for a domain that may have other entries (e.g. MX records, SOA record, etc). If a CNAME record exists for a particular domain, all other records may be ignored (using the word "may" here because it is explicitly disallowed in the RFC but implementations seem to differ). [Reference](https://www.freecodecamp.org/news/why-cant-a-domain-s-root-be-a-cname-8cbab38e5f5c/).

Having 2nd level domains is straightforward (e.g. `isomer.sg`), because that is the minimum number you can go, and we know for sure it has to be a root domain. For this case, this check is valid. For a root domain, we should create an A record to ensure compliance with the RFC (side note: The `type` in line 241 may have to be adjusted).

Having 3rd and above level domains is less straightforward (e.g. `isomer.gov.sg` or `bendemeersec.moe.edu.sg`). Based on this check, `isomer.gov.sg` would not be a root domain and a CNAME record would be created. However, this would conflict with the existing MX records present for the domain (which allows `admin@isomer.gov.sg` to exist). The same applies for some schools, like `temasekjc.moe.edu.sg` which is a 4th level domain, but there are MX records defined for it. Creating a CNAME record for this domain may cause the MX records to be ignored, which is the behaviour that the RFC requires.

Basically, this means that it is possible for different domains on the same level to be treated differently. For instance, `www.isomer.gov.sg` and `temasekjc.moe.edu.sg` are both 4th level domains, but `www.isomer.gov.sg` is not a root domain whereas `temasekjc.moe.edu.sg` is a root domain.

I can think of 2 possible solutions:
1. Easier solution: Utilise the inputs given in the site launch form to determine whether to use the A or CNAME record. The primary domain is the one that will use the CNAME record, and the redirection domain is the one that will use the A record plus any additional records like MX record, etc.
2. Tougher solution: Perform a DNS lookup query to see if there are any existing records for the given domain. If there is, then it has to be a root domain (i.e. cannot use CNAME record here, have to use an A record). Otherwise, it is not a root domain and CNAME record is fine.

Sorry for the long post! What do you think about this?